### PR TITLE
Add support for multi-target read and writes

### PR DESCRIPTION
**Pull request by arnikola**

### CODE REVIEW

File: src/cmd/promremotebench/query.go [hunk 1]
```diff
@@ -237,24 +245,83 @@ func (q *queryExecutor) alertLoad(checker Checker) {
...
+		return results, nil
+	}
+
+	firstResult := results[0]
+	for i, res := range results[1:] {
+		if bytes.Equal(res, firstResult) {
+			continue
+		}
+
+		q.Logger.Error("mismatch in returned data", zap.Int("index", i))
```
Comment (notbdu) [inline src/cmd/promremotebench/query.go]: Should we log the mismatching data here?

File: src/cmd/promremotebench/main.go [hunk 2]
```diff
@@ -70,13 +71,23 @@ const (
 	maxNumScrapesActive = 4
 )
 
+type targetUrls []string
+
+func (u *targetUrls) String() string {
+	return "a slice of target urls"
+}
+
+func (u *targetUrls) Set(value string) error {
```
Comment (notbdu) [inline src/cmd/promremotebench/main.go]: nit: Would `Append` be better here? `Set` sounds like a full set and replace op.

File: src/cmd/promremotebench/main.go [hunk 3]
```diff
@@ -108,12 +118,24 @@ func main() {
...
+	var writeTargetURLs targetUrls
+	flag.Var(&writeTargetURLs, "target", "Target remote write endpoint (for remote write)")
+
+	// Can have multiple query-targets.
+	var queryTargetURLs targetUrls
+	flag.Var(&queryTargetURLs, "query-target", "Target query endpoint(s) (for exercising by proxy remote read)")
 	flag.Parse()
-	if len(*targetURL) == 0 {
+
+	if len(writeTargetURLs) == 0 {
```
Comment (notbdu) [inline src/cmd/promremotebench/main.go]: The default behaviour before would also set a default target url of `http://localhost:7201/receive` although I don't think it really matters. It's likely that these targets always need to be set.

File: src/cmd/promremotebench/main.go (hunk 2 above)
Comment (arnikola) [reply src/cmd/promremotebench/main.go]: Unfortunately it has to be `Set` to satisfy `flag.Value`

File: src/cmd/promremotebench/main.go (hunk 3 above)
Comment (arnikola) [reply src/cmd/promremotebench/main.go]: Yeah it was a little weird before that it had default behaviour and then overrode it only if they'd supplied something (but only failed on nil?) I'll just set it to default if empty :+1:

File: src/cmd/promremotebench/query.go (hunk 1 above)
Comment (arnikola) [reply src/cmd/promremotebench/query.go]: Not initially, I think; they're likely to be huge byte buffers which won't really be readable anyway. If we start seeing this failing a lot, maybe we could parse them out into actual query results for debugging, but this is probably fine for now

File: src/cmd/promremotebench/main.go (hunk 2 above)
Comment (arnikola) [reply src/cmd/promremotebench/main.go]: Added a comment to that effect 👍

**Pull request merged**